pandas/tseries/index.py

```diff
@@ -77,16 +77,19 @@ def f(self):
 
             result = tslib.get_start_end_field(values, field, self.freqstr,
                                                month_kw)
+            result = self._maybe_mask_results(result, convert='float64')
+
```

Comment:
the problem is with the weekday_name, which gives strings, and for this the `astype('float64')` will fail